ref(debug-files): Split into smaller pages (#2878)
getsentry/sentry-docs: src/platforms/common/data-management/debug-files/file-formats.mdx

---
title: "Debug File Formats"
sidebar_order: 1

> description: "Learn about the platform-specific file formats and the debug information they contain."

Reviewer: ```suggestion
description: "Learn about platform-specific file formats and the debug information they contain."
```

---

Sentry differentiates four kinds of debug information:

- **Debug Information:** Provides function names, paths to source files, line

>   numbers and inline frames. The process of resolving this information from

Reviewer: ```suggestion
  numbers, and inline frames. The process of resolving this information from
```

```

This can be verified by checking for the `C` flag in `readelf`, corresponding to
`SHF_COMPRESSED`:

```bash
readelf -S path/to/file
  ...
  [21] .debug_info       PROGBITS         0000000000000000  00000370
       000000000000e133  0000000000000000   C       0     0     1
```

## PE and PDB

On Microsoft Windows, executables and dynamic libraries use the _Portable
Executable_ container, in short _PE_. Debug information is stored in _Program
Database_ files, commonly referred to as _PDB_.

- **Executables** use an `.exe` file extension. Only when compiled for 64-bit
  architectures, they contain unwind information. Otherwise, they do not contain
  any usable information and will be omitted for the upload to Sentry.

- **Dynamic Libraries** use the `.dll` file extension, and otherwise behave
  exactly the same as executables.

- **Debug Companions** are stored in `.pdb` files. They usually contain debug
  information and in most cases symbol tables. For 32-bit programs, they also
  contain unwind information. In rare cases, they might have different file
  names than their corresponding executable.

<Note>

> At the moment, Sentry supports Native PDBs only. PDBs for the .NET platform are not supported.

Reviewer: ```suggestion
Currently, Sentry supports only Native PDBs. PDBs for the .NET platform are not supported.
```

</Note>

## Breakpad Symbols

The Google Breakpad library has established a platform-independent ASCII format
to store debug information. Such files are usually generated for applications

> using Breakpad, Crashpad or the Electron Framework.

Reviewer: ```suggestion
using Breakpad, Crashpad, or the Electron Framework.
```
